config.go

```diff
@@ -0,0 +1,30 @@
+package fyneselfupdate
+
+import (
+	"crypto/ed25519"
+	"time"
+
+	"fyne.io/fyne/v2"
+	"github.com/fynelabs/selfupdate"
+)
+
+// NewConfig create a selfupdate.Config structure that provide Fyne UI dialog to progress the update
+func NewConfig(app fyne.App, win fyne.Window, source selfupdate.Source, schedule selfupdate.Schedule, publicKey ed25519.PublicKey) *selfupdate.Config {
```

Comment:
Aren't schedule and publicKey optional?
If so let's leave them off here, constructors should be mandatory really.

Reply:
PublicKey definitively not. I could make schedule optional, but it isn't at the moment. What would be the expected default behavior if there is no schedule set?

Reply:
Hmm, not sure about missing schedule - maybe just check on start?
Maybe it's not optional.
My concern about the key is that this API is now tied to `ed25519` - I think it should be `crypto.PublicKey`, that way this is much more flexible.

Reply:
Being tied to ed25519 is a opinionated choice here. I think if any alternate signature mechanism should be provided, they should be explicit and part of a new API or a new breaking version of the API.